lib/ember-plugins.js

```diff
@@ -0,0 +1,167 @@
+        },
+      },
+      "@ember/application/deprecations stripping",
+    ],
+  ];
+}
+function _emberVersionRequiresModulesAPIPolyfill() {
+  // once a version of Ember ships with the
+  // emberjs/rfcs#176 modules natively this will
+  // be updated to detect that and return false
+  return true;
+}
+
+function _getEmberModulesAPIPolyfill(appRoot, config) {
+  if (config.disableEmberModulesAPIPolyfill) {
+    return;
+  }
+
+  if (_emberVersionRequiresModulesAPIPolyfill()) {
+    const ignore = _getEmberModulesAPIIgnore(appRoot, config);
+
+    return [
+      [require.resolve("babel-plugin-ember-modules-api-polyfill"), { ignore }],
+    ];
+  }
+}
+
+function _shouldIgnoreEmberString(appRoot) {
+  return resolvePackagePath("@ember/string", appRoot) !== null;
+}
+
+function _shouldIgnoreJQuery(appRoot) {
+  let packagePath = resolvePackagePath("@ember/jquery", appRoot);
+  if (packagePath === null) {
+    return true;
+  }
+  let pkg = require(packagePath);
+  return pkg && semver.gt(pkg.version, "0.6.0");
+}
+
+function _emberDataVersionRequiresPackagesPolyfill(appRoot) {
+  let packagePath = resolvePackagePath("ember-data", appRoot);
+  if (packagePath === null) {
+    return false;
+  }
+  let pkg = require(packagePath);
+  return pkg && semver.lt(pkg.version, "3.12.0-alpha.0");
+}
+
+function _getEmberModulesAPIIgnore(appRoot, config) {
+  const ignore = {
+    "@ember/debug": ["assert", "deprecate", "warn"],
+    "@ember/application/deprecations": ["deprecate"],
+  };
+
+  if (config.shouldIgnoreEmberString || _shouldIgnoreEmberString(appRoot)) {
+    ignore["@ember/string"] = [
+      "fmt",
+      "loc",
+      "w",
+      "decamelize",
+      "dasherize",
+      "camelize",
+      "classify",
+      "underscore",
+      "capitalize",
+      "setStrings",
+      "getStrings",
+      "getString",
+    ];
+  }
+  if (config.shouldIgnoreJQuery || _shouldIgnoreJQuery(appRoot)) {
+    ignore["jquery"] = ["default"];
+  }
+
+  return ignore;
+}
+
+function _getEmberDataPackagesPolyfill(appRoot, config) {
+  if (config.emberDataVersionRequiresPackagesPolyfill) {
+    return [[require.resolve("babel-plugin-ember-data-packages-polyfill")]];
+  }
+  return _emberDataVersionRequiresPackagesPolyfill(appRoot);
+}
+
+function _getModuleResolutionPlugins(config) {
+  if (!config.disableModuleResolution) {
+    const resolvePath = require("../lib/relative-module-paths")
+    .resolveRelativeModulePath;
+    return [
+      [require.resolve("babel-plugin-module-resolver"), { resolvePath }],
+      [
+        require.resolve("@babel/plugin-transform-modules-amd"),
+        { noInterop: true },
+      ],
+    ];
+  }
+}
+
+function _getProposalDecoratorsAndClassPlugins(config) {
+  if (!config.shouldIgnoreDecoratorAndClassPlugins) {
+    return [
+      ["@babel/plugin-proposal-decorators", { legacy: true }],
+      ["@babel/plugin-proposal-class-properties"],
+    ];
+  }
+}
+
+module.exports = function (appRoot, config = {}) {
```

Comment:
Can you add some API docs for this file that list the possible options that are allowed (as it is, it's kinda hard to see the full list of possible options)?